File: src/tests/DataBlockASPNETDemo/vb.net/mappings/BookAuthors.vb
Imports System
Imports System.Data
Imports voidsoft.DataBlock
Namespace BookManager.Mappings

	<Serializable()> _
	Public Class BookAuthors
	Inherits TableMetadata
		Private _fields As DatabaseField()

		Public Sub New()
            _fields = New DatabaseField(2) {}
			_fields(0) = New DatabaseField(DbType.Int32, "Id", True, True, Nothing)
			_fields(1) = New DatabaseField(DbType.Int32, "AuthorId", False, False, Nothing)
			_fields(2) = New DatabaseField(DbType.Int32, "BookId", False, False, Nothing)
			Me.currentTableName = "BookAuthors"
		End Sub

		Public Overloads Overrides Property TableFields() As DatabaseField()
			Get
				Return _fields
			End Get
			Set
				_fields = value
			End Set
		End Property

		Public Property Id() As System.Int32
			Get
				Return CType((Me.GetField("Id")).fieldValue, System.Int32)
			End Get
			Set
				Me.SetFieldValue("Id", value)
			End Set
		End Property

		Public Property AuthorId() As System.Int32
			Get
				Return CType((Me.GetField("AuthorId")).fieldValue, System.Int32)
			End Get
			Set
				Me.SetFieldValue("AuthorId", value)
			End Set
		End Property

		Public Property BookId() As System.Int32
			Get
				Return CType((Me.GetField("BookId")).fieldValue, System.Int32)
			End Get
			Set
				Me.SetFieldValue("BookId", value)
			End Set
		End Property
	End Class 
End Namespace

File: src/tests/DataBlockASPNETDemo/csharp/Mappings/BookAuthors.cs
using System;
using System.Data;
using voidsoft.DataBlock;

namespace BookManager.Mappings
{

    [Serializable()]
    public class BookAuthors : TableMetadata
    {
        private DatabaseField[] _fields;

        public BookAuthors()
        {
            _fields = new DatabaseField[3];
            _fields[0] = new DatabaseField(DbType.Int32, "Id", true, true, null);
            _fields[1] = new DatabaseField(DbType.Int32, "AuthorId", false, false, null);
            _fields[2] = new DatabaseField(DbType.Int32, "BookId", false, false, null);

            this.currentTableName = "BookAuthors";


        }


        public override DatabaseField[] TableFields
        {
            get
            {
                return _fields;
            }
            set
            {
                _fields = value;
            }
        }

        public System.Int32 Id
        {
            get
            {
                return (System.Int32)(this.GetField("Id")).fieldValue;
            }

            set
            {
                this.SetFieldValue("Id", value);
            }
        }


        public System.Int32 AuthorId
        {
            get
            {
                return (System.Int32)(this.GetField("AuthorId")).fieldValue;
            }

            set
            {
                this.SetFieldValue("AuthorId", value);
            }
        }


        public System.Int32 BookId
        {
            get
            {
                return (System.Int32)(this.GetField("BookId")).fieldValue;
            }

            set
            {
                this.SetFieldValue("BookId", value);
            }
        }

    }
}
